### PR TITLE
[MI-2310] Resolve TODO comments

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/components/admin_settings/__snapshots__/download_button.test.tsx.snap [hunk 1]
```diff
@@ -12,7 +12,7 @@ exports[`Download Button Should render correctly 1`] = `
   >
     <a
       download={true}
-      href="http://localhost:8065/plugins/mattermost-plugin-servicenow/api/v1/download"
+      href="http://localhost:8065/plugins/mattermost-plugin-servicenow/public/servicenow_for_mattermost_notifications_v2.0.xml"
```
Comment (manojmalik20) [inline webapp/src/components/admin_settings/__snapshots__/download_button.test.tsx.snap:15]: We should create a constant for the filename, I think.

File: webapp/tsconfig.json [hunk 2]
```diff
@@ -20,8 +20,12 @@
     "noEmit": true,
     "experimentalDecorators": true,
     "jsx": "react",
-    "baseUrl": "./src",
+    "baseUrl": ".",
```
Comment (manojmalik20) [inline webapp/tsconfig.json:23]: What was the need for this?

File: webapp/src/plugin_constants/index.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ export enum ToggleSwitchLabelPositioning {
     Right = 'right',
 }
 
-export const DOWNLOAD_UPDATE_SET_LINK = '/download';
+export const UPLOAD_SET_FILE = '/servicenow_for_mattermost_notifications_v2.0.xml';
```
Comment (manojmalik20) [inline webapp/src/plugin_constants/index.ts:9]:
```suggestion
export const UPLOAD_SET_FILENAME = 'servicenow_for_mattermost_notifications_v2.0.xml';
```

File: webapp/src/utils/index.tsx [hunk 4]
```diff
@@ -6,21 +6,23 @@ import React from 'react';
...
     pluginApiBaseUrl: string;
     mattermostApiBaseUrl: string;
+    uploadSetFile: string;
 } => {
     const url = new URL(window.location.href);
     const baseUrl = `${url.protocol}//${url.host}`;
     const pluginUrl = `${baseUrl}/plugins/${pluginId}`;
     const pluginApiBaseUrl = `${pluginUrl}/api/v1`;
     const mattermostApiBaseUrl = `${baseUrl}/api/v4`;
+    const uploadSetFile = `${baseUrl}/plugins/${pluginId}/public`;
```
Comment (manojmalik20) [inline webapp/src/utils/index.tsx:23]:
```suggestion
    const publicFilesUrl = `${pluginUrl}/public`;
```

File: webapp/src/components/admin_settings/__snapshots__/download_button.test.tsx.snap (hunk 1 above)
Comment (ayusht2810) [reply webapp/src/components/admin_settings/__snapshots__/download_button.test.tsx.snap:15]: this is a snapshot file, I don't think we should manually edit it

File: webapp/tsconfig.json (hunk 2 above)
Comment (ayusht2810) [reply webapp/tsconfig.json:23]: this is done to remove the typescript errors (`Cannot find module`) coming from node modules

File: webapp/src/components/admin_settings/__snapshots__/download_button.test.tsx.snap (hunk 1 above)
Comment (manojmalik20) [reply webapp/src/components/admin_settings/__snapshots__/download_button.test.tsx.snap:15]: Ohh.. my bad.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**